mapstore-mongodb/src/main/scala/net/spals/appbuilder/mapstore/mongodb/MongoDatabaseProvider.scala

```diff
@@ -17,13 +17,27 @@ private[mongodb] class MongoDatabaseProvider @Inject() (
   @ApplicationName applicationName: String,
   mongoClient: MongoClient
 ) extends Provider[MongoDatabase] {
+  private val LOGGER = LoggerFactory.getLogger(classOf[MongoDatabaseProvider])
 
-  @NotNull
   @Configuration("mapStore.mongoDB.database")
   @volatile
+  @VisibleForTesting
   private[mongodb] var configuredDatabaseName: String = null
 
-  private lazy val databaseName = Option(configuredDatabaseName).getOrElse(applicationName)
+  @Configuration("mapStore.mongoDB.url")
+  @volatile
+  @VisibleForTesting
+  private[mongodb] var url: String = null
+
+  @VisibleForTesting
+  private[mongodb] lazy val databaseName = {
+    val mongoClientURI = Option(url).map(new MongoClientURI(_))
+    mongoClientURI.map(_.getDatabase)
+      .getOrElse(Option(configuredDatabaseName).getOrElse(applicationName))
```

Comment:
Check for a database in the URL first. If present, the client will already be configured with the database name so this will match. If not present, fallback to the `mapStore.mongoDB.url` value and if that doesn't exist, fallback to the application name.